### PR TITLE
Clean up built-in List type

### PR DESCRIPTION
**Pull request by ttuegel**
---

###### Review checklist

The author performs the actions on the checklist. The reviewer evaluates the work and checks the boxes as they are completed.

-   [ ] **Summary.** Write a summary of the changes. Explain what you did to fix the issue, and why you did it. Present the changes in a logical order. Instead of writing a summary in the pull request, you may push a clean Git history.
-   [ ] **Documentation.** Write documentation for new functions. Update documentation for functions that changed, or complete documentation where it is missing.
-   [ ] **Tests.** Write unit tests for every change. Write the unit tests that were missing before the changes. Include any examples from the reported issue as integration tests.
-   [ ] **Clean up.** The changes are already clean. Clean up anything near the changes that you noticed while working. This does not mean only spatially near the changes, but logically near: any code that interacts with the changes!

### CODE REVIEW

File: kore/src/Kore/Internal/InternalList.hs [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    synthetic = fold
+    {-# INLINE synthetic #-}
+
+-- | An 'InternalList' pattern is always 'Function'.
+instance Synthetic Function InternalList where
+    synthetic = fold
+    {-# INLINE synthetic #-}
+
+-- | An 'InternalList' pattern is always 'Functional'.
+instance Synthetic Functional InternalList where
```
Comment (ana-pantilie) [inline kore/src/Kore/Internal/InternalList.hs:106]: I don't think these comments are right.

File: kore/src/Kore/Step/Simplification/InternalList.hs [hunk 2]
```diff
@@ -0,0 +1,32 @@
+{- |
+Copyright   : (c) Runtime Verification, 2019
```
Comment (ana-pantilie) [inline kore/src/Kore/Step/Simplification/InternalList.hs:2]:
```suggestion
Copyright   : (c) Runtime Verification, 2020
```

File: kore/src/Kore/Internal/InternalList.hs [hunk 3]
```diff
@@ -0,0 +1,112 @@
+{- |
+Copyright   : (c) Runtime Verification, 2019
```
Comment (ana-pantilie) [inline kore/src/Kore/Internal/InternalList.hs:2]:
```suggestion
Copyright   : (c) Runtime Verification, 2020
```

**Review by ana-pantilie — COMMENTED** [review]
I noticed that the changes to the integration tests were that the final patterns are now simplified conditionally. Is there any reason for this, or should we not worry about this change?

**Comment by ttuegel** [comment]
> I noticed that the changes to the integration tests were that the final patterns are now simplified conditionally. Is there any reason for this, or should we not worry about this change?

I did not notice that, thanks for pointing it out! I don't think it's right.

**Comment by ttuegel** [comment]
@ana-pantilie This is ready to be reviewed again.

**Review by ana-pantilie — APPROVED** [review]

**Pull request merged**